src/scenegraph.ts

```diff
@@ -77,184 +76,69 @@ export type Scenegraph = {
   [key: Id]: ScenegraphNode;
 };
 
-// Propagates bbox dims to other dims that can be inferred
-export function propagateBBoxValues(bbox: BBox, owners: BBoxOwners): void {
-  // const inferredBBox = { ...bbox };
+export type ScenegraphElement = {
+  jsx: JSX.Element;
+  layout: (parentId: Id | null) => void;
+};
 
-  inferenceRules.forEach((rule) => {
-    if (
-      rule.from.every((key) => bbox[key] !== undefined) &&
-      bbox[rule.to] === undefined
-    ) {
-      bbox[rule.to] = rule.calculate(rule.from.map((key) => bbox[key]!));
-      owners[rule.to] = inferred;
-    }
-  });
-}
+export const resolveScenegraphElements = (
```

Comment:
added a comment

Reply:
The comment helps a bit. I understand what children of a component means in the sense of `props.children` but isn't that always an array? (or a signal of an array?) I still don't understand why `unresolved` could be undefined, null, an array, or an object.

Also, what does it mean to pass info up a JSX tree? Where is bluefish passing this info up? Does it get passed to components as a prop? How does `SceneGraphElement` relate to `ScenegraphNode`?

Maybe I'm opening a can of worms here. Just wanted to document my further questions here.

Reply:
`props.children` is a `JSX.Element`, which can be an array of children or a literal, similar to React. eg if you pass a numeric literal it will be a numeric literal in the true. same for null. it can be a function if it calls a signal (in which case it is auto-wrapped to preserve reactivity).

By pass info up I mean literally return it to its parent. The props.children ultimately resolve to an array of objects rather than normal elements.

Reply:
This solidjs behavior is not super well-documented...